### ISSUE TITLE
Start on boot issue on pfSense 2.4.4

### ISSUE THREAD

**Issue by Fail-Safe** [nextdns/nextdns#37]
After testing the freebsd_amd64 v1.3.3 client on pfSense 2.4.4 for several days, I am happy to report it runs very well. However, after a reboot today I realized the NextDNS client did not start. I rebooted again to verify and confirmed the same.

After investigating, I found that pfSense does not follow the standard implementation of 'rc' and therefore any third-party scripts in `/usr/local/etc/rc.d/` must end in a .sh extension in order to execute at boot: [shell-script-option](https://docs.netgate.com/pfsense/en/latest/development/executing-commands-at-boot-time.html#shell-script-option)

I tried a couple things so I could report back the behavior:

1. [Simple "fix" 1] Added `/usr/local/bin/nextdns start` as an [earlyshellcmd option](https://docs.netgate.com/pfsense/en/latest/development/executing-commands-at-boot-time.html#earlyshellcmd-option). This worked perfectly and it still allowed me to execute `nextdns <start|stop|restart|status>` commands from the shell as expected.
2. [Simple "fix" 2] Added `/usr/local/bin/nextdns start` as a [shellcmd option](https://docs.netgate.com/pfsense/en/latest/development/executing-commands-at-boot-time.html#shellcmd-option). This worked perfectly and will likely be my path forward as it is **recommended to use this over `earlyshellcmd` when possible**. It still allowed me to execute `nextdns <start|stop|restart|status>` commands from the shell as expected.
3. Removed `earlyshellcmd` and `shellcmd` as referenced above and renamed `/usr/local/etc/rc.d/nextdns` to `/usr/local/etc/rc.d/nextdns.sh`. This executed the script properly and started the service upon boot as desired, but broke the ability to execute `nextdns ...` commands at the shell:
```
[2.4.4-RELEASE][root@pfSense-n1.xxx.yyy]/root: nextdns status
Error: the service is not installed
[2.4.4-RELEASE][root@pfSense-n1.xxx.yyy]/root: service nextdns status
nextdns does not exist in /etc/rc.d or the local startup
directories (/usr/local/etc/rc.d), or is not executable
```
Listing services (`service -l | grep nextdns`) revealed it was now known as `nextdns.sh`, so the following worked:
```
[2.4.4-RELEASE][root@pfSense-n1.xxx.yyy]/root: service nextdns.sh status
nextdns is running as pid 63866.
```

So, given the options at this point, option 2 will be my path forward. It allows me to continue acting upon the NextDNS client as designed and won't require extra steps if the client is upgraded (or uninstalled/reinstalled for testing reasons). If there are better options I have not considered, I'm all ears.

I wanted to make you aware of this behavior on pfSense, however, just in case you wanted to add some additional verbiage in the README.md for those using pfSense as well.

**Comment by mrancier** [comment]
I am attempting to install this on pfSense 2.4.4p3 and every time I try to run it on the command line I get: No service system detected.

Any idea how to get around this ?

**Comment by rs** [comment]
What version do you have?

**Comment by mrancier** [comment]
Latest

**Comment by mrancier** [comment]
nextdns version dev

**Comment by mrancier** [comment]
That is what the version arg says

**Comment by mrancier** [comment]
[2.4.4-RELEASE][admin@pfSense.RANFAMICOM.lan.local]/root: /usr/local/bin/nextdns start
2019/12/27 12:19:19 No service system detected.

**Comment by mrancier** [comment]
Same with run

**Comment by mrancier** [comment]
/usr/local/bin/nextdns run
2019/12/27 12:35:41 No service system detected.

**Comment by mrancier** [comment]
Fuck all then, eh ?  Thanks anyway, thought I've give it shot.

**Comment by Fail-Safe** [comment]
@mrancier Hold your horses a little bit there. Just because you jumped into the middle of your project doesn't mean everyone else's worlds joined your orbit at the same time. :-) The devs on this project are reasonably responsive, so have a little patience.

So did you install the service or are you just trying to run it without installing the service?

**Comment by mrancier** [comment]
Sorry 'bout that.  tryin' to be funny and fell flat.  Precompiled binary works, but the service always fails with status 1.  So, yes, I was just trying to run a build from source and that happened.  Not sure about the service issue.

**Comment by mrancier** [comment]
Error: Failed to start NextDNS Proxy: exit status 1

**Comment by Fail-Safe** [comment]
Out of curiosity, could you download and try the v1.3.3 I was using. I'm curious if something changed after that build and you're hitting a new issue.

**Comment by mrancier** [comment]
Nevermind.  1.3.5 started working.  Problem is apparent only on dev.  Thanks.

**Comment by rs** [comment]
The dev version means you built it from source right? Apparently you built it on an older revision.

Stay tuned tho, next revision will provide a better pfsense support.

**Comment by gregbinsd** [comment]
rs, I will be waiting for the next revision to the pfSense port, as well.  I chatted with you 20 min ago on your main website.  I have 1 production pfSense 2.4.4-p3 system and 2 sandbox pfSense systems.   I am looking forward to using NextDNS, since I have been having troubles with the other options available.  I will be happy to run beta tests for you on my systems.
Greg

**Comment by rs** [comment]
Fix by v1.4.0
